lib/rpmvercmp.c

```diff
@@ -45,6 +45,20 @@ int rpmvercmp(const char * a, const char * b)
 	    continue;
 	}
 
+	/*
+	 * Handle caret separator. Concept is same as tilde, except if one of
+	 * strings ends (base version), another considered as higher version.
```

Comment:
"... Concept is the same as tilde, except that if one of the strings ends (base version), the other is considered as higher version." (just grammar)